sharding/networksharding/peerShardMapper.go

```diff
@@ -235,27 +229,24 @@ func (psm *PeerShardMapper) getPeerInfoSearchingPidInFallbackCache(pid core.Peer
 }
 
 // GetLastKnownPeerID returns the newest updated peer id for the given public key
-func (psm *PeerShardMapper) GetLastKnownPeerID(pk []byte) (*core.PeerID, bool) {
-	psm.mutUpdatePeerIdPublicKey.RLock()
```

Comment:
no need for the RLock anymore? the pkPeerIdCache is updated in the updatePeerIDPublicKey method

Reply:
the protection is now done internally in the pid queue handler + the mutex in the cacher (supposing we have one)
The lock found only on updatePeerIDPublicKey ensures the method is executed sequentially, as in a critical section to be able to tell 100% that those write/modify operations are done sequentially (otherwise we can get WAR or WAW hazards)

Reply:
ok